server/src/com/thoughtworks/go/server/messaging/plugin/PluginNotificationListenerFactory.java

```diff
@@ -0,0 +1,49 @@
+/*************************GO-LICENSE-START*********************************
+ * Copyright 2014 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *************************GO-LICENSE-END***********************************/
+
+package com.thoughtworks.go.server.messaging.plugin;
+
+import com.thoughtworks.go.plugin.access.notification.NotificationExtension;
+import com.thoughtworks.go.serverhealth.ServerHealthService;
+import com.thoughtworks.go.util.SystemEnvironment;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PluginNotificationListenerFactory {
+    private PluginNotificationQueue pluginNotificationQueue;
+    private SystemEnvironment systemEnvironment;
+    private NotificationExtension notificationExtension;
+    private final ServerHealthService serverHealthService;
+
+    @Autowired
+    public PluginNotificationListenerFactory(PluginNotificationQueue pluginNotificationQueue, SystemEnvironment systemEnvironment,
+                                             NotificationExtension notificationExtension, ServerHealthService serverHealthService) {
+        this.pluginNotificationQueue = pluginNotificationQueue;
+        this.systemEnvironment = systemEnvironment;
+        this.notificationExtension = notificationExtension;
+        this.serverHealthService = serverHealthService;
+    }
+
+    public void init() {
+        int numberOfListeners = systemEnvironment.getNumberOfPluginNotificationListener();
+
+        for (int i = 0; i < numberOfListeners; i++) {
+            PluginNotificationService pluginNotificationService = new PluginNotificationService(notificationExtension, serverHealthService);
```

Comment:
on similar lines as [this](https://github.com/gocd/gocd/blob/master/server/src/com/thoughtworks/go/server/materials/MaterialUpdateListenerFactory.java#L77). Need to check why it isn't autowired here as well.